internal/controllers/device_data_controller.go

```diff
@@ -407,6 +408,71 @@ func (udc *UserDevicesController) GetUserDeviceErrorCodeQueries(c *fiber.Ctx) er
 	return c.JSON(GetUserDeviceErrorCodeQueriesResponse{Queries: queries})
 }
 
+// ClearUserDeviceErrorCodeQuery godoc
+// @Description Clear user device error codes
+// @Tags        user-devices
+// @Success     200 {object} controllers.QueryDeviceErrorCodesResponse
+// @Security    BearerAuth
+// @Router      /user/devices/{userDeviceID}/error-codes/clear [post]
+func (udc *UserDevicesController) ClearUserDeviceErrorCodeQuery(c *fiber.Ctx) error {
+	userID := helpers.GetUserID(c)
+	udi := c.Params("userDeviceID")
+
+	logger := helpers.GetLogger(c, udc.log)
+
+	userDeviceExists, err := models.UserDevices(
+		models.UserDeviceWhere.UserID.EQ(userID),
+		models.UserDeviceWhere.ID.EQ(udi),
+	).Exists(c.Context(), udc.DBS().Reader)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return fiber.NewError(fiber.StatusNotFound, err.Error())
+		}
+
+		logger.Err(err).Msg("error occurred when fetching error codes for device")
+		return fiber.NewError(fiber.StatusInternalServerError, "error occurred updating device error queries")
+	}
+
+	if !userDeviceExists {
+		return fiber.NewError(fiber.StatusUnauthorized, "error, user does not own device")
+	}
+
```

Comment:
Are you imagining that they could go into an an historical query and clear that? Physically but I don’t think it’s a, I don’t think it makes sense.

Let’s ask mobile.

Reply:
So on the UI i can see a list and a clear button, which tells me they can choose to select what to clear. What bothers me is if they can choose what to clear, then clearing based on the last created error code wouldn't make sense in that case